src/main/scala/com.snowplowanalytics/sqs2kinesis/EventsStreamModule.scala

```diff
@@ -0,0 +1,119 @@
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.sqs2kinesis
+
+import com.github.matsluni.akkahttpspi.AkkaHttpClient
+import software.amazon.awssdk.services.sqs.SqsAsyncClient
+import software.amazon.awssdk.services.sqs.model.Message
+import com.amazonaws.services.kinesis.AmazonKinesisAsyncClientBuilder
+import com.amazonaws.services.kinesis.model.PutRecordsResultEntry
+import com.amazonaws.services.kinesis.model.PutRecordsRequestEntry
+import java.nio.ByteBuffer
+import akka.actor.ActorSystem
+import akka.stream.scaladsl._
+import akka.stream.alpakka.sqs.scaladsl.SqsSource
+import akka.stream.alpakka.sqs.SqsSourceSettings
+import akka.stream.alpakka.kinesis.KinesisFlowSettings
+import akka.stream.alpakka.kinesis.scaladsl.KinesisFlow
+import akka.stream.alpakka.sqs.MessageAction
+import akka.stream.alpakka.sqs.scaladsl.SqsAckSink
+import akka.NotUsed
+import akka.stream.ActorAttributes
+import akka.stream.Supervision
+import scala.concurrent.duration._
+import com.typesafe.scalalogging.Logger
+
+object EventsStreamModule {
+
+  case class StreamConfig(
+    sqsQueue: String,
+    sqsKeyValueSeparator: Char,
+    kinesisStreamName: String
+  )
+
+  val logger = Logger[EventsStreamModule.type]
+
+  type KinesisKeyAndMsg = (String, ByteBuffer)
+
+  def sqsMsg2kinesisMsg(sqsKeyValueSeparator: Char): String => KinesisKeyAndMsg =
+    msgBody => {
+      val decoded          = java.util.Base64.getDecoder().decode(msgBody)
+      val (key, msg)       = decoded.splitAt(decoded.indexOf(sqsKeyValueSeparator.toByte))
+      val withoutSeparator = msg.tail
+      (new String(key), ByteBuffer.wrap(withoutSeparator))
+    }
+
+  def runStream(config: StreamConfig)(implicit system: ActorSystem) = {
+
+    implicit val sqsClient = {
+      val client = SqsAsyncClient.builder().httpClient(AkkaHttpClient.builder().withActorSystem(system).build()).build()
+
+      system.registerOnTermination(client.close())
+      client
+    }
+
+    implicit val kinesisClient = {
+      val client = AmazonKinesisAsyncClientBuilder.standard().build()
+
+      system.registerOnTermination(client.shutdown())
+      client
+    }
+
+    val sqsSource: Source[Message, NotUsed] =
+      SqsSource(config.sqsQueue, SqsSourceSettings.Defaults)
+
+    val toPutRecordReqEntry: KinesisKeyAndMsg => PutRecordsRequestEntry = {
```

Comment:
Why not defining it with `def` ?

Reply:
What are you asking about? 

Why it is not `def toPutRecordReqEntry(....): .... `
I defined all the functions in this module like that to be consistent with `Flow`: `val kinesisFlow: Flow[(PutRecordsRequestEntry, Message), (PutRecordsResultEntry, Message), NotUsed] =`

or 

Why it's not `def toPutRecordReqEntry: ... => ....`

because it's not necessary

Reply:
I always use `def` for functions and `val` for values and I was curious whether there is a good reason to use `val` instead of `def` for functions or is it just a matter of taste?

Reply:
I think it's a matter of taste/consistency. I'm also using `def` for most of the times, here it's consistent to use `val` and `def` with this definition would create the function more than one time potentially, so it's better to use `val` if we have the former style of definition.